app/controllers/comments_controller.rb

```diff
@@ -6,14 +6,19 @@ def new
   end
 
   def create
-    @user = User.find(params[:user_id])
+    @user = current_user
     @post = Post.find(params[:post_id])
     @comment = @post.comments.new(comment_params)
     @comment.author = @user
-    if @comment.save
-      redirect_to request.referrer
-    else
-      render :new
+
+    respond_to do |format|
+      if @comment.save
+        format.html { redirect_to request.referrer, notice: 'Comment was successfully created.' }
+        format.json { render Json: @comment, status: :created }
+      else
+        format.html { render :new }
+        format.json { render json: { errors: @comment.errors.full_messages }, status: :unprocessable_entity }
+      end
```

Comment:
- [ ] Good job working on this but please kindly create an API endpoint to list all comments for a user's post.

Reply:
@AnselemOdims thanks for the feedback , we have been added endpoint for displaying comment  for a given post as JSON format